pkg/indexer/nft.go

```diff
@@ -116,14 +206,78 @@ func (i *Indexer) NFTOutputsWithFilters(filters ...NFTFilterOption) *IndexerResu
 		query = query.Where("address = ?", addr[:])
 	}
 
-	if opts.requiresDustReturn != nil {
-		if *opts.requiresDustReturn {
+	if opts.hasDustReturnCondition != nil {
+		if *opts.hasDustReturnCondition {
 			query = query.Where("dust_return IS NOT NULL")
 		} else {
 			query = query.Where("dust_return IS NULL")
 		}
 	}
 
+	if opts.dustReturnAddress != nil {
+		addr, err := addressBytesForAddress(*opts.dustReturnAddress)
+		if err != nil {
+			return errorResult(err)
+		}
+		query = query.Where("dust_return_address = ?", addr[:])
+	}
+
+	if opts.hasExpirationCondition != nil {
+		if *opts.hasExpirationCondition {
+			query = query.Where("expiration_return_address IS NOT NULL")
+		} else {
+			query = query.Where("expiration_return_address IS NULL")
+		}
+	}
+
+	if opts.expirationReturnAddress != nil {
+		addr, err := addressBytesForAddress(*opts.expirationReturnAddress)
+		if err != nil {
+			return errorResult(err)
+		}
+		query = query.Where("expiration_return_address = ?", addr[:])
+	}
+
+	if opts.expiresBefore != nil {
+		query = query.Where("expiration_time > ?", *opts.expiresBefore)
+	}
+
+	if opts.expiresAfter != nil {
+		query = query.Where("expiration_time < ?", *opts.expiresAfter)
+	}
+
+	if opts.expiresBeforeMilestone != nil {
+		query = query.Where("expiration_milestone > ?", *opts.expiresBeforeMilestone)
+	}
+
+	if opts.expiresAfterMilestone != nil {
+		query = query.Where("expiration_milestone < ?", *opts.expiresAfterMilestone)
+	}
+
+	if opts.hasTimelockCondition != nil {
+		if *opts.hasTimelockCondition {
+			query = query.Where("(timelock_time IS NOT NULL OR timelock_milestone IS NOT NULL)")
+		} else {
+			query = query.Where("timelock_time IS NULL").Where("timelock_milestone IS NULL")
+		}
+	}
+
+	if opts.timelockedBefore != nil {
+		query = query.Where("timelock_time > ?", *opts.timelockedBefore)
+	}
+
+	if opts.timelockedAfter != nil {
+		query = query.Where("timelock_time < ?", *opts.timelockedAfter)
+	}
+
+	if opts.timelockedBeforeMilestone != nil {
+		query = query.Where("timelock_milestone > ?", *opts.timelockedBeforeMilestone)
+	}
+
+	if opts.timelockedAfterMilestone != nil {
+		query = query.Where("timelock_milestone < ?", *opts.timelockedAfterMilestone)
```

Comment:
Isn't it the other way around?
```suggestion
		query = query.Where("timelock_milestone > ?", *opts.timelockedAfterMilestone)
```